docs/guides/getting-started/getting-started-scaling-and-tuning.adoc

```diff
@@ -0,0 +1,62 @@
+
+If you are experiencing timeouts in obtaining database connections, you should consider increasing the number of connections available - see <@links.ha id="concepts-database-connections" />.
+
+=== Vertical Autoscaling
+
+Some platforms, such as Kubernetes, provide mechanisms to vertically autoscale. That is not recommended for {project_name} if it requires restarting the server instance - which is currently the case for Java on Kubernetes. You can consider instead providing higher CPU and/or memory limits to allow your JVM to adapt within those limits as needed.
+
+== Horizontal Scaling
+
+A single {project_name} instance is susceptible to availability issues. If the instance goes down you will be faced with a full outage until another instance comes up, and unless the '`persistent-user-sessions`' feature is enabled all of your ongoing sessions will be lost. By running at least two cluster members on different machines you will greatly increase the availability of {project_name}.
+
+There will also be a limit on the amount of concurrent requests a single JVM can effectively handle. Additional server instances can provide roughly linear scaling of throughput until associated resources, such as the database or distributed caching, limit scaling.
+
+In general, it is best to let the {project_name} Operator handle horizontal scaling concerns. When using the Operator simply set the Keycloak custom resource `spec.instances` as desired to horizontally scale - see <@links.ha id="deploy-keycloak-kubernetes" /> for more.
+
+If you are not using the Operator, please review the following:
+
+* Your instances should be on separate machines if possible to increase availability. On Kubernetes you can use Pod anti-affinitity to enforce this.
+
+* You need to use distributed caching and for multi-site clusters external caching for cluster members to share the same session state - see <@links.server id="caching" /> for more on the relevant configuration. The embedded Infinispan cache has horizontal scaling considerations including:
+
+- Your instances will need a way to find each other, see discovery in the server guide.
+- It is not well suited for clusters spanning multiple availability zones, also known as a stretch clusters. It is recommended that when using the embedded Infinispan cache that the instances all exist in the same availability zone to avoid unnecessary round-trips in the communication that would amplify in the response times. On Kubernetes you can use Pod affinity to enforce this grouping of Pods.
+- It does not gracefully handle multiple members joining or leaving at the same time - the latter can lead to data loss. On Kubernetes you can use a StatefulSet with the default serial handling to ensure Pods are started and stopped sequentially.
+
+To avoid losing service availability when a whole site is not available, see the high availability guide for more information on a multi-site deployment - see <@links.ha id="introduction" />.
```

Comment:
```suggestion
To avoid losing service availability when a whole site is unavailable, see the high availability guide for more information on a multi-site deployment - see <@links.ha id="introduction" />.
```